apps/coscrad-frontend/src/components/resources/songs/song-index.presenter.tsx

```diff
@@ -1,21 +1,37 @@
-import { AggregateType, ISongViewModel } from '@coscrad/api-interfaces';
+import { AggregateType, IMultilingualText, ISongViewModel } from '@coscrad/api-interfaces';
+import { AudioClipPlayer } from '@coscrad/media-player';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import { LinkOff } from '@mui/icons-material';
+import { useContext } from 'react';
+import { ConfigurableContentContext } from '../../../configurable-front-matter/configurable-content-provider';
 import { SongIndexState } from '../../../store/slices/resources/songs/types';
 import { HeadingLabel, IndexTable } from '../../../utils/generic-components/presenters/tables';
+import { Matchers } from '../../../utils/generic-components/presenters/tables/generic-index-table-presenter/filter-table-data';
 import { CellRenderersDefinition } from '../../../utils/generic-components/presenters/tables/generic-index-table-presenter/types/cell-renderers-definition';
 import { renderAggregateIdCell } from '../utils/render-aggregate-id-cell';
-import { formatBilingualText } from '../vocabulary-lists/utils';
+import { renderMultilingualTextCell } from '../utils/render-multilingual-text-cell';
+
+export const SongIndexPresenter = (songsIndexResult: SongIndexState) => {
+    const { defaultLanguageCode } = useContext(ConfigurableContentContext);
+
+    const { entities: songs } = songsIndexResult;
 
-export const SongIndexPresenter = ({ entities: songs }: SongIndexState) => {
     const headingLabels: HeadingLabel<ISongViewModel>[] = [
-        { propertyKey: 'title', headingLabel: 'Title' },
-        { propertyKey: 'titleEnglish', headingLabel: 'English' },
-        // { propertyKey: 'audioURL', headingLabel: 'Audio' },
+        { propertyKey: 'id', headingLabel: 'Link' },
+        { propertyKey: 'name', headingLabel: 'Song' },
+        { propertyKey: 'audioURL', headingLabel: 'Audio' },
     ];
 
     const cellRenderersDefinition: CellRenderersDefinition<ISongViewModel> = {
-        title: renderAggregateIdCell,
-        titleEnglish: ({ titleEnglish, title }) => formatBilingualText(titleEnglish, title),
-        // audioURL: ({ audioURL }: ISongViewModel) => audioURL,
+        id: renderAggregateIdCell,
+        name: ({ name }: ISongViewModel) => renderMultilingualTextCell(name, defaultLanguageCode),
+        audioURL: ({ audioURL }: ISongViewModel) =>
+            isNullOrUndefined(audioURL) ? <LinkOff /> : <AudioClipPlayer audioUrl={audioURL} />,
+    };
+
+    const matchers: Matchers<ISongViewModel> = {
+        name: ({ items }: IMultilingualText, search) =>
```

Comment:
Do you need to search by language? We might want to work on this and break out some helpers because every resource index view will have a name column.